src/string.cr

```diff
@@ -3314,7 +3314,7 @@ class String
           if mem
             if char == '_'
               # case 3
-            else
+            elsif upcase || downcase
```

Comment:
You intended `last_is_upcase` instead of `upcase` correctly, didn't you?

This is not the perfect though. For example, `"3.14IsPi"` should be converted to `"3.14_is_pi"`. But your implementation converts `"3.14is_pi".